src/libraries/System.Runtime/tests/System/Numerics/GenericMathDimHelpers.cs

```diff
@@ -0,0 +1,98 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+
+using System.Globalization;
+using System.Numerics;
+using System.Runtime.InteropServices;
+using Xunit;
+
+namespace System.Numerics.Tests
+{
+
+    public struct BinaryNumberDimHelper : IBinaryNumber<BinaryNumberDimHelper>
+    {
+        public int value = 0;
+        BinaryNumberDimHelper(int value)
+        {
+            this.value = value;
+        }
+        static BinaryNumberDimHelper INumberBase<BinaryNumberDimHelper>.Zero => new BinaryNumberDimHelper(0);
+
+        static BinaryNumberDimHelper IBitwiseOperators<BinaryNumberDimHelper, BinaryNumberDimHelper, BinaryNumberDimHelper>.operator ~(BinaryNumberDimHelper value)
+        {
+            return new BinaryNumberDimHelper(~value.value);
+        }
```

Comment:
nit: Fix newlines and add `private` before the constructor:
```suggestion
    public struct BinaryNumberDimHelper : IBinaryNumber<BinaryNumberDimHelper>
    {
        public int Value = 0;

        private BinaryNumberDimHelper(int value)
        {
            Value = value;
        }

        static BinaryNumberDimHelper INumberBase<BinaryNumberDimHelper>.Zero => new BinaryNumberDimHelper(0);

        static BinaryNumberDimHelper IBitwiseOperators<BinaryNumberDimHelper, BinaryNumberDimHelper, BinaryNumberDimHelper>.operator ~(BinaryNumberDimHelper value)
        {
            return new BinaryNumberDimHelper(~value.value);
        }
```